msgpack_pack.c

```diff
@@ -280,15 +280,15 @@ static inline void msgpack_serialize_array(smart_str *buf, zval *val, HashTable
 					value_noref = value;
 				}
 
-				if ((Z_TYPE_P(value_noref) == IS_ARRAY && ZEND_HASH_GET_APPLY_COUNT(Z_ARRVAL_P(value_noref)) > 1)) {
+				if (Z_TYPE_P(value_noref) == IS_ARRAY && Z_IS_RECURSIVE_P(value_noref)) {
```

Comment:
I think the test failures may be related to this part.
Z_IS_RECURSIVE is mostly equiv. to ZEND_HASH_GET_APPLY_COUNT > 0 (not > 1)

Reply:
Thanks for the review! What do you recommend to bridge this gap? Is there a different check that will say if a single count is present?

Reply:
Sadly, I have no idea, probably @laruence will know

Reply:
Interesting - the in-tree JSON extension is using `GC_IS_RECURSIVE` instead of `Z_IS_RECURSIVE`: https://github.com/php/php-src/blob/master/ext/json/json_encoder.c#L143

Reply:
I was able to fix the yaml ext which have the same issue 
https://github.com/php/pecl-file_formats-yaml/pull/33

But this require a bit more change... need to find the proper fix.